code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm

```diff
@@ -753,280 +753,174 @@
-	if(windup)
-		if(!do_after(warden, windup, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, numticks = 1))
-			to_chat(warden, SPAN_XENOWARNING("We cancel our retrieve."))
-			apply_cooldown()
 
-			for (var/obj/effect/xenomorph/xeno_telegraph/XT in telegraph_atom_list)
-				telegraph_atom_list -= XT
-				qdel(XT)
 
-			REMOVE_TRAIT(warden, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Praetorian Retrieve"))
+	apply_cooldown()
+	return ..()
 
-			return
+/datum/action/xeno_action/onclick/fight_or_flight/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/valkyrie_flight = owner
 
-	REMOVE_TRAIT(warden, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Praetorian Retrieve"))
+	var/datum/behavior_delegate/praetorian_valkyrie/behavior = valkyrie_flight.behavior_delegate
+	var/image/effect_overlay = get_busy_icon(ACTION_PURPLE_POWER_UP)
 
-	playsound(get_turf(warden), 'sound/effects/bang.ogg', 25, 0)
 
-	var/successful_retrieve = FALSE
-	for(var/turf/target_turf in turflist)
-		if(targetXeno in target_turf)
-			successful_retrieve = TRUE
-			break
+	if (!valkyrie_flight.check_state())
+		return
 
-	if(!successful_retrieve)
-		to_chat(warden, SPAN_XENOWARNING("We can't reach [targetXeno] with our resin retrieval hook!"))
+	if(!action_cooldown_check())
 		return
 
-	to_chat(targetXeno, SPAN_XENOBOLDNOTICE("We are pulled toward [warden]!"))
+	if (!behavior.use_internal_fury_ability(rejuvenate_cost))
+		return
 
-	shake_camera(targetXeno, 10, 1)
-	var/throw_dist = get_dist(throw_target_turf, targetXeno)-1
-	if(throw_target_turf == behind_turf)
-		throw_dist++
-		to_chat(warden, SPAN_XENOBOLDNOTICE("We fling [targetXeno] over our head with our resin hook, and they land behind us!"))
+	if (behavior.base_fury < 100)
+		playsound(valkyrie_flight, 'sound/voice/xenos_roaring.ogg', 125)
+		for(var/mob/living/carbon/xenomorph/allied_xenomorphs in range(low_rage_range, valkyrie_flight))
+			to_chat(allied_xenomorphs, SPAN_XENOWARNING("Every single inch in our body moves on its own to fight."))
+			valkyrie_flight.create_shriekwave(3)
+			allied_xenomorphs.xeno_jitter(1 SECONDS,)
+			allied_xenomorphs.flick_heal_overlay(3 SECONDS, "#F5007A")
+			effect_overlay.flick_overlay(allied_xenomorphs, 20)
 	else
-		to_chat(warden, SPAN_XENOBOLDNOTICE("We fling [targetXeno] towards us with our resin hook, and they land in front of us!"))
-	targetXeno.throw_atom(throw_target_turf, throw_dist, SPEED_VERY_FAST, pass_flags = PASS_MOB_THRU)
+		playsound(valkyrie_flight, 'sound/voice/xenos_roaring.ogg', 125)
+		for(var/mob/living/carbon/xenomorph/allied_xenomorphs in range(high_rage_range, valkyrie_flight))
+
+			to_chat(allied_xenomorphs, SPAN_XENOWARNING("Every single inch in our body moves on its own to fight."))
+			valkyrie_flight.create_shriekwave(3)
+			allied_xenomorphs.xeno_jitter(1 SECONDS)
+			allied_xenomorphs.flick_heal_overlay(3 SECONDS, "#F5007A")
+			effect_overlay.flick_overlay(allied_xenomorphs, 20)
```

Comment:
There is nothing altered between these two blocks other than the first argument to `range`.

Do something like
`var/range = behavior.base_fury < 100 ? low_rage_range : high_rage_range`
and then just have one block rather than an if/else.